backend/dataall/modules/datasets/handlers/s3_location_handler.py

```diff
@@ -0,0 +1,48 @@
+import logging
+
+from dataall.aws.handlers.service_handlers import Worker
+from dataall.aws.handlers.sts import SessionHelper
+from dataall.db import models
+from dataall.modules.datasets.services.dataset_location import DatasetLocationService
+
+log = logging.getLogger(__name__)
+
+
+class S3DatasetLocationHandler:
+    """Handles async requests related to s3 for dataset storage location"""
+
+    @staticmethod
+    def client(account_id: str, region: str, client_type: str):
```

Comment:
Please separate the "client" relevant functions into a different file, as you did for the table in pt1: https://github.com/awslabs/aws-dataall/pull/413/files#diff-968c8c65b5e3316a5bdc465f52a4ac90189835df927b176faadf992daa43eb2f

Reply:
Good notice! Thanks!